src/main/java/com/epam/aidial/core/upstream/WeightedRoundRobinBalancer.java

```diff
@@ -0,0 +1,112 @@
+package com.epam.aidial.core.upstream;
+
+import com.epam.aidial.core.config.Upstream;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import javax.annotation.Nullable;
+
+@Getter
+@Slf4j
+public class WeightedRoundRobinBalancer implements Comparable<WeightedRoundRobinBalancer>, LoadBalancer<UpstreamState> {
+
+    private final int tier;
+    private final List<UpstreamState> upstreams;
```

Comment:
Should we distribute load across upstreams in proportion to their weights?

For example there are 3 upstreams with weights: 2, 3 and 5. That means if there are 10 requests each upstream will take 2, 3 and 5 requests respectively. Right?

So we treat each weight as 20%, 30% and 50% in the example above.